lib/matplotlib/cbook/__init__.py

```diff
@@ -2148,16 +2148,6 @@ def _check_and_log_subprocess(command, logger, **kwargs):
     return report
 
 
-def _check_not_matrix(**kwargs):
```

Comment:
This was only used in `_parse_args`, we don't check anywhere else in the code for `np.matrix`. 

 `np.matrix` itself is [not recommended anymore](https://docs.scipy.org/doc/numpy/reference/generated/numpy.matrix.html) and [matplotlib usage: Types of inputs](https://matplotlib.org/tutorials/introductory/usage.html#types-of-inputs-to-plotting-functions) states that `np.matrix` may or may not work.

Therefore I don't see the need to explicitly check for this to give an error message.

Reply:
This was added in #13089 to fix #1558 just in 3.1.0; do we really want to revert that check so quickly?

Also, skipping X and Y was [done on purpose](https://github.com/matplotlib/matplotlib/pull/7461#issuecomment-262613003), though perhaps that's a little unintuitive.